libbeat/common/schema/mapstriface/mapstriface.go

```diff
@@ -98,23 +98,23 @@ func (convMap ConvMap) Map(key string, event common.MapStr, data map[string]inte
 		err.Required = convMap.Required
 		return multierror.Errors{err}
 	}
-	subData, ok := d.(map[string]interface{})
-	if !ok {
+	switch subData := d.(type) {
+	case map[string]interface{}, common.MapStr:
+		subEvent := common.MapStr{}
+		_, errors := convMap.Schema.ApplyTo(subEvent, subData.(map[string]interface{}))
+		for _, err := range errors {
+			if err, ok := err.(schema.KeyError); ok {
+				err.SetKey(convMap.Key + "." + err.Key())
+			}
+		}
+		event[key] = subEvent
+		return errors
+	default:
 		msg := fmt.Sprintf("expected dictionary, found %T", subData)
```

Comment:
Would it be the same to use `d` here?
```suggestion
		msg := fmt.Sprintf("expected dictionary, found %T", d)
```

Reply:
It will be the same output.